cmd/kube-proxy/app/server.go

```diff
@@ -237,7 +264,31 @@ func (o *Options) Run() error {
 		return err
 	}
 
-	return proxyServer.Run()
+	errCh := make(chan error)
+	defer close(errCh)
+
+	// run the proxy in goroutine
+	go func() {
+		err := proxyServer.Run()
+		errCh <- err
+	}()
+
+	for {
+		select {
+		case err := <-o.watcherError:
+			errCh <- err
```

Comment:
rather than reading the error here and below and then writing to a channel, why not just `return err` ?